device/futhark/src/cca.cpp

```diff
@@ -0,0 +1,70 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2021-2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include <memory>
+#include <traccc/futhark/cca.hpp>
+#include <traccc/futhark/utils.hpp>
+#include <traccc/futhark/wrapper.hpp>
+#include <vecmem/containers/vector.hpp>
+#include <vecmem/memory/host_memory_resource.hpp>
+#include <vector>
+
+namespace traccc::futhark {
+component_connection::output_type component_connection::operator()(
+    const cell_container_types::host &data) const {
+    std::size_t total_cells = 0;
+
+    for (std::size_t i = 0; i < data.size(); ++i) {
+        total_cells += data.at(i).items.size();
+    }
+
+    std::vector<uint64_t> host_event(total_cells);
+    std::vector<uint64_t> host_geometry(total_cells);
+    std::vector<int64_t> host_channel0(total_cells);
+    std::vector<int64_t> host_channel1(total_cells);
+    std::vector<float> host_activation(total_cells);
+
+    for (std::size_t i = 0, k = 0; i < data.size(); ++i) {
+        for (std::size_t j = 0; j < data.at(i).items.size(); ++j, ++k) {
+            host_event[k] = 0;
+            host_geometry[k] = data.at(i).header.module;
+            host_channel0[k] = data.at(i).items.at(j).channel0;
+            host_channel1[k] = data.at(i).items.at(j).channel1;
+            host_activation[k] = data.at(i).items.at(j).activation;
+        }
+    }
+
+    cells_to_measurements_wrapper::output_t r =
+        cells_to_measurements_wrapper::run(
+            std::move(host_event), std::move(host_geometry),
+            std::move(host_channel0), std::move(host_channel1),
+            std::move(host_activation));
+
+    output_type out;
+    vecmem::host_memory_resource mem;
```

Comment:
I would prefer to make the algorithm receive the memory resource that it should use, from the outside. So that a user could for instance set it up to put its output into SYCL shared memory. Or HIP host memory. Or some cached version of any of those...

In general when I see short-live memory resource objects, that tends to be a bad code smell for me. Even if creating `vecmem::host_memory_resource` on demand is of course very cheap.